src/dbnode/storage/entry.go

```diff
@@ -138,14 +133,24 @@ func (entry *Entry) IndexedBlockCount() int {
 }
 
 // IndexedForBlockStart returns a bool to indicate if the Entry has been successfully
-// indexed for the given index blockstart.
+// indexed for the given index blockStart.
 func (entry *Entry) IndexedForBlockStart(indexBlockStart xtime.UnixNano) bool {
 	entry.reverseIndex.RLock()
 	isIndexed := entry.reverseIndex.indexedWithRLock(indexBlockStart)
 	entry.reverseIndex.RUnlock()
 	return isIndexed
 }
 
+// IndexedRange returns minimum and maximum blockStart values covered by index entry.
+// Note that there may be uncovered gaps within that range.
+// Returns (0, 0) for an empty range.
+func (entry *Entry) IndexedRange() (xtime.UnixNano, xtime.UnixNano) {
+	entry.reverseIndex.RLock()
+	defer entry.reverseIndex.RUnlock()
```

Comment:
Might want to remove the defer (e.g. like above method) since it is faster. Only worthwhile in hot paths.